ui/apps/platform/src/services/DeploymentsService.ts

```diff
@@ -52,10 +44,46 @@ export function fetchDeployments(
     if (query) {
         queryObject.query = query;
     }
-    const params = queryString.stringify(queryObject, { arrayFormat: 'repeat', allowDots: true });
+    return queryString.stringify(queryObject, { arrayFormat: 'repeat', allowDots: true });
+}
+
+/**
+ * Fetches list of registered deployments.
+ *
+ * Changes from the 'fetchDeploymentsLegacy' function:
+ * - uses the new `SearchFilter` type instead of `RestSearchOption`
+ * - Does not fetch process information linked to the deployment
+ */
+export function listDeployments(
+    searchFilter: SearchFilter,
+    sortOption: Record<string, string>,
+    page: number,
+    pageSize: number
+): Promise<ListDeployment[]> {
```

Comment:
As a follow up after achieving the access control goal, we might test whether `CancellableRequest` is needed to prevent memory leaks if someone selects namespaces, and then immediately clicks a link in left navigation to leave NetworkGraph page.